### PR TITLE
Adjust translog after versionType is removed in 7.0

### PR DESCRIPTION
**Pull request by dnhatn**
With the presence of sequence number, we no longer use versionType to
resolve out of order collision and remove it in 7.0.

This PR adjusts translog to adapt that change.

Relates #31945

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/translog/Translog.java [hunk 1]
```diff
@@ -1250,7 +1254,11 @@ private Delete(final StreamInput in) throws IOException {
                 id = uidObject.id();
             }
             this.version = in.readLong();
-            this.versionType = VersionType.fromValue(in.readByte());
+            if (format <= FORMAT_SEQ_NO) {
+                this.versionType = VersionType.fromValue(in.readByte());
+            } else {
+                this.versionType = VersionType.EXTERNAL; // versionType is removed in 7.0
```
Comment (bleskes) [inline server/src/main/java/org/elasticsearch/index/translog/Translog.java]: I wonder if it's worth while to back port the format identifier here .. at very least add a comment and assert that we will never use the format id of 7.0 here if we end up changing the format in the 6.x life time.

**Review by bleskes — APPROVED** [review]
LGTM. Left one request

**Comment by dnhatn** [comment]
@bleskes I've "backported" the format versions from 7.0. Can you please take another look?

**Comment by dnhatn** [comment]
Thanks @bleskes

**Pull request merged**